pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionWorkerStarter.java

```diff
@@ -66,11 +66,12 @@ public static void main(String[] args) throws Exception {
         try {
             worker.start();
         } catch (Throwable th) {
+            log.error("Encountered error in function worker.", th);
             worker.stop();
             System.exit(-1);
```

Comment:
System.exit() is not always guaranteed to take the process down (eg: hanging threads). Use `Runtime.halt()` instead